### PR TITLE
Ported Responses from Simple

### PR DESCRIPTION
**Pull request by koahmad98**

### CODE REVIEW

File: lio-http-server/src/LIO/HTTP/Server/Responses.hs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  ( ok, okHtml, okJson, okXml
+  , movedTo, redirectTo
+  , badRequest, requireBasicAuth, forbidden
+  , notFound
+  , serverError
+  ) where
+
+import qualified Data.ByteString.Char8 as S8
+import qualified Data.ByteString.Lazy.Char8 as L8
+import Network.HTTP.Types
```
Comment (deian) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: do we need this? I think LIO.HTTP.Server exports most of these?

File: lio-http-server/src/LIO/HTTP/Server/Responses.hs [hunk 2]
```diff
@@ -0,0 +1,148 @@
+{-# LANGUAGE Trustworthy #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Taken from Simple
```
Comment (deian) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: I think we can leave the simple comment without the `-- |` so it doesn't get generated in the haddock html output

Comment (deian) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: If you haven't already: `stack haddock` and open the generated file

File: lio-http-server/src/LIO/HTTP/Server/Responses.hs [hunk 3]
```diff
@@ -0,0 +1,148 @@
+{-# LANGUAGE Trustworthy #-}
```
Comment (deian) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: Have you tried `Safe` first? It's okay either way

File: lio-http-server/src/LIO/HTTP/Server/Responses.hs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+             , L8.pack url
+             , L8.pack "\">here</A>\n\
+                       \</BODY></HTML>\n"]
+
+-- | Given a URL returns a 303 (See Other) 'Response' redirecting to that URL.
+redirectTo :: S8.ByteString -> Response
+redirectTo url = mkHtmlResponse status303 [(hLocation, url)] html
+  where html = L8.concat
+             [L8.pack
+              "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n\
```
Comment (deian) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: mind just double checking to see if anything has changed due to html5 ;)?

Comment (koahmad98) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: So the doctype is for HTML 2.0, I think it's done on purpose to support as many browsers as possible. Do you still want me to switch it to HTML 5?

File: lio-http-server/src/LIO/HTTP/Server/Responses.hs (hunk 1 above)
Comment (koahmad98) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: removed the network import, the ByteStrings are still needed

File: lio-http-server/src/LIO/HTTP/Server/Responses.hs (hunk 2 above)
Comment (koahmad98) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: done

File: lio-http-server/src/LIO/HTTP/Server/Responses.hs (hunk 3 above)
Comment (koahmad98) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: Safe didn't work, kept Trustworthy

Comment (deian) [inline lio-http-server/src/LIO/HTTP/Server/Responses.hs]: What flagged it?

File: lio-http-server/examples/HelloWorld.hs [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
 import LIO.HTTP.Server
+import LIO.HTTP.Server.Responses
 
 main :: IO ()
 main = server 3000 "127.0.0.1" app
 
 app :: DCApplication
-app _ = do
-  return $ Response status200 [] "Hello World!"
+app _ = return (okHtml "Hello World!")
```
Comment (deian) [inline lio-http-server/examples/HelloWorld.hs]: can you just create another file? Or add another app. We want both: ability to create responses manually and use the responses API. Adding additional examples, again even if in same file, that exercise other APIs exposed by lib would be great.

**Pull request merged**